v2/api/v2/egress_types.go

```diff
@@ -119,13 +130,97 @@ func (es EgressSpec) validate() field.ErrorList {
 		}
 	}
 
+	if es.PodDisruptionBudget != nil {
+		pp := p.Child("podDisruptionBudget")
+		allErrs = append(allErrs, validatePodDisruptionBudgetSpec(*es.PodDisruptionBudget, pp)...)
+	}
+
 	return allErrs
 }
 
 func (es EgressSpec) validateUpdate() field.ErrorList {
 	return es.validate()
 }
 
+// For validating PodDisruptionBudgetSpec.
+// Original code is in https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/policy/validation/validation.go
+
+type UnhealthyPodEvictionPolicyType string
+
+const (
+	IfHealthyBudget UnhealthyPodEvictionPolicyType = "IfHealthyBudget"
+	AlwaysAllow     UnhealthyPodEvictionPolicyType = "AlwaysAllow"
+)
+
+var supportedUnhealthyPodEvictionPolicies = sets.NewString(
+	string(IfHealthyBudget),
+	string(AlwaysAllow),
+)
+
+func validatePodDisruptionBudgetSpec(spec policyv1.PodDisruptionBudgetSpec, fldPath *field.Path) field.ErrorList {
```

Comment:
I think the check may be omitted from here, because it will be tested anyway when we create a PDB.

Reply:
We decided to implement validation for `minAvailable` and `maxUnavailable`.